hmacauth.go

```
@@ -114,60 +120,56 @@ func stringToSign(req *http.Request, options *Options, timestamp string) (string
 	return buffer.String(), nil
 }
 
-func parseAuthHeader(header string) (AuthBits, error) {
-
-	var ab AuthBits
+func parseAuthHeader(header string) (*authBits, error) {
+	if header == "" {
+		return nil, HMACAuthError{"Authorization Header Not Supplied"}
```

Let's return an error here instead of allowing empty strings or making ourselves check for content before submitting.